### PR TITLE
Support creating a DataChannel during renegotation

### LINKED ISSUE #1361: DataChannel is not created if a consistent Track already exists
### Your environment.
  - Browser: Google Chrome 84.0.4147.105
  - Other Information - *stacktraces, related issues, suggestions how to fix, links for us to have context*

### What did you do?

Pion: CreateTrack -> Negotiation -> CreateDataChannel -> Negotiation 

### What did you expect?
In offer don't have application  section 

### What happened?
In offer have application  section 

Offer after CreateTrack
```sdp
v=0
o=- 5921791100870032632 1597257153 IN IP4 0.0.0.0
s=-
t=0 0
a=fingerprint:sha-256 2E:44:4D:56:ED:E2:DE:41:22:57:5D:D9:D7:DF:02:D9:E0:75:F2:55:CF:FF:DD:C4:57:C1:E0:9D:7F:45:8F:FC
a=group:BUNDLE 0
m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:0
a=ice-ufrag:kEHhadmyXpKuBIdO
a=ice-pwd:UFskHDIfAYisgaxTZaJKrLmMugHMgJbR
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:96 VP8/90000
a=rtpmap:98 VP9/90000
a=rtpmap:102 H264/90000
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:2596996162 cname:video-2854263694
a=ssrc:2596996162 msid:video-2854263694 video-4039455774
a=ssrc:2596996162 mslabel:video-2854263694
a=ssrc:2596996162 label:video-4039455774
a=msid:video-2854263694 video-4039455774
a=sendrecv
```

Offer after CreateDataChannel
```sdp
v=0
o=- 6170903342722614613 1597257163 IN IP4 0.0.0.0
s=-
t=0 0
a=fingerprint:sha-256 2E:44:4D:56:ED:E2:DE:41:22:57:5D:D9:D7:DF:02:D9:E0:75:F2:55:CF:FF:DD:C4:57:C1:E0:9D:7F:45:8F:FC
a=group:BUNDLE 0
m=video 9 UDP/TLS/RTP/SAVPF 96 98 102
c=IN IP4 0.0.0.0
a=setup:actpass
a=mid:0
a=ice-ufrag:kEHhadmyXpKuBIdO
a=ice-pwd:UFskHDIfAYisgaxTZaJKrLmMugHMgJbR
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:96 VP8/90000
a=rtpmap:98 VP9/90000
a=rtpmap:102 H264/90000
a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f
a=ssrc:2596996162 cname:video-2854263694
a=ssrc:2596996162 msid:video-2854263694 video-4039455774
a=ssrc:2596996162 mslabel:video-2854263694
a=ssrc:2596996162 label:video-4039455774
a=msid:video-2854263694 video-4039455774
a=sendrecv
a=candidate:foundation 1 udp 2130706431 192.168.61.2 57898 typ host generation 0
a=candidate:foundation 2 udp 2130706431 192.168.61.2 57898 typ host generation 0
a=candidate:foundation 1 udp 2130706431 192.168.108.145 57899 typ host generation 0
a=candidate:foundation 2 udp 2130706431 192.168.108.145 57899 typ host generation 0
a=candidate:foundation 1 udp 1694498815 88.88.88.88 6677 typ srflx raddr 0.0.0.0 rport 57900 generation 0
a=candidate:foundation 2 udp 1694498815 88.88.88.88 6677 typ srflx raddr 0.0.0.0 rport 57900 generation 0
a=end-of-candidates
```

### PR DESCRIPTION
**Pull request by Sean-Der**
Before when creating an offer during renegotation we didn't support
adding a DataChannel.

Resolves #1361

### CODE REVIEW

File: peerconnection.go [hunk 1]
```diff
@@ -1856,8 +1856,9 @@ func (pc *PeerConnection) generateUnmatchedSDP(useIdentity bool) (*sdp.SessionDe
 		if len(audio) > 0 {
 			mediaSections = append(mediaSections, mediaSection{id: "audio", transceivers: audio})
 		}
-		for i := uint32(0); i < pc.sctpTransport.dataChannelsRequested; i++ {
```
Comment (Sean-Der) [inline peerconnection.go:1859]: This was incorrect before. We only want one media section, this would be create one for every DataChannel.

**Comment by tekig** [comment]
Section appeared, but the channel does not open in Chrome and Firefox

**Comment by Sean-Der** [comment]
Oh nice catch, let me update test and fix that as well @tekig !

That is easy as well. Just multiple code paths in SetRemoteDescription :)

**Comment by Sean-Der** [comment]
@tekig sorry about the delay! Mind trying again?

**Pull request merged**